Infrastructure/Infrastructure/Utilities/ExcelTools.cs

```diff
@@ -14,26 +14,24 @@ public static DataSet ReadAsDataSet(string file)
         {
             var format = Path.GetExtension(file);
 
-            using (var stream = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
-            {
-                IWorkbook book;
+            using var stream = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.ReadWrite);
```

Comment:
Are you sure this is right?
I thought the idea of the _using_ statement is that it cleans up after itself?

Reply:
Are you talking about the lack of closed braces following the _using_ statement? Because if so that's a new syntax option available (I think from C# 9) that the IDE kept pestering me to use. It functions exactly the same, just one less set of messy braces.

Reply:
Yep, that's what I was talking about.
So how does it know where the last bracket would be then?
Can you screen grab the hint from the IDE?

Reply:
Or find a reference somewhere that explains what it is doing and how it works now.

Reply:
Here's the prompt I'm seeing:

![image](https://user-images.githubusercontent.com/53413135/120135771-510ef000-c214-11eb-96e6-2349d340f827.png)

And the [documentation](https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-8#using-declarations) for the feature (from C# 8 apparently).

Reply:
I did a quick test, and it only makes that suggestion for _using_ blocks that encompass entire methods. When I added test code outside the block, the prompt disappeared. So if the method wasn't returning at the end of the _using_ block I would have to include the braces.

Reply:
which probably means the using is superfluous as it has a return inside it

Reply:
I don't think so, the way I've used the using statement is almost identical to the example I linked in the documentation. My understandstanding is that returning from the method won't necessarily release unmanaged resources like file streams because the GC doesn't know how to properly handle them. It's recommended to explicitly release such objects with _using_ or _Dispose()_ as soon as you're finished with them.

Reply:
Then I would move the return statement outside of the using block.
Are you checking if the file exists before it gets to this code?

Reply:
The file is taken from an OpenFileDialog, so it has to exist to reach that point in the code. 
And that's the advantage of the new style of _using_. It works out from context when it's free to release the object, so you don't have to worry about what to include/exclude from the statement block.

Reply:
No, it doesn't work out the context - the compiler replaces the using statement with the try...finally block at the end of it's scope now, not immediately after the end of the block statement as it did before.
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/using-statement

Reply:
Ahhh, my bad, I didn't realise using statements were compiled as try-finally blocks. But in that case moving the return location shouldn't make a difference right? Because a finally block executes as soon as you leave the try block, including when return is called, returning inside one is the same as calling return immediately afterwards - in both cases the finally block disposes the object and then the value is returned to the calling statement which continues execution. Or is there a distinction that I'm missing?

Reply:
The new format will work in this case - just checking you understood why it works before merging it.